inference-engine/src/transformations/src/transformations/op_conversions/simplify_ctc_greedy_decoder.cpp

```diff
@@ -0,0 +1,98 @@
+        auto transpose_seq_mask = std::make_shared<ngraph::opset6::Transpose>(seq_mask->output(0),
+                                                                              ngraph::opset6::Constant::create(seq_len_type,
+                                                                                                               Shape({2}), {1, 0}));
+        auto transpose_seq_mask_f32 = std::make_shared<ngraph::opset6::Convert>(transpose_seq_mask->output(0), element::f32);
+        auto simplified_decoder = std::make_shared<ngraph::opset6::CTCGreedyDecoder>(transpose,
+                                                                                     transpose_seq_mask_f32->output(0),
+                                                                                     decoder_v6->get_merge_repeated());
+        simplified_decoder->set_friendly_name(decoder_v6->get_friendly_name());
+
+        auto squeeze2_axis = ngraph::opset6::Constant::create(seq_len_type, Shape({1}), {3});
+        auto squeeze2_output_f = std::make_shared<ngraph::opset6::Squeeze>(simplified_decoder->output(0), squeeze2_axis);
+        auto squeeze1_axis = ngraph::opset6::Constant::create(seq_len_type, Shape({1}), {2});
+        auto squeeze1_output_f = std::make_shared<ngraph::opset6::Squeeze>(squeeze2_output_f->output(0), squeeze1_axis);
+
+        element::Type ci_type = decoder_v6->get_classes_index_type();
+        element::Type sl_type = decoder_v6->get_sequence_length_type();
+        auto output_i = std::make_shared<ngraph::opset6::Convert>(squeeze1_output_f->output(0), ci_type);
+        auto minus1 = opset6::Constant::create(sl_type, Shape{}, {-1});
+        auto where_equal_minus1 = std::make_shared<ngraph::opset6::Equal>(output_i, minus1);
+
+        auto seq_mask_const0 = opset6::Constant::create(sl_type, Shape{1}, {0});
+        auto seq_mask_const1 = opset6::Constant::create(sl_type, Shape{1}, {1});
+        auto output_seq_mask = std::make_shared<ngraph::opset6::Select>(where_equal_minus1, seq_mask_const0, seq_mask_const1);
+        auto seq_mask_axis = opset6::Constant::create(seq_len_type, Shape{1}, {1});
+        auto output_seq_len = std::make_shared<ngraph::opset6::ReduceSum>(output_seq_mask, seq_mask_axis);
+        ngraph::copy_runtime_info(decoder_v6, {transpose, simplified_decoder, data_shape, T, N, plusT, plusT_scalar, range1T, mask_shape, upper_bounds,
+                                               squeeze2_output_f, squeeze1_output_f, transpose_upper_bounds, bool_seq_mask, seq_mask, transpose_seq_mask,
+                                               transpose_seq_mask_f32, output_i, where_equal_minus1, output_seq_mask, output_seq_len});
+
+        output_i->set_friendly_name(decoder_v6->get_friendly_name()+".0");
+        output_seq_len->set_friendly_name(decoder_v6->get_friendly_name()+".1");
+        ngraph::replace_node(decoder_v6, {output_i->output(0), output_seq_len->output(0)});
```

Comment:
Please add detailed description about the logic in this pass. I see no comments at all.
You can find comment reference here: https://github.com/openvinotoolkit/openvino/pull/4038/files#diff-122cf9d33ca4e95fcea2eeb8c6897ded37854b1e8b9929e0c1d8f030e136ca25R19